### PR TITLE
feat: allow stacks-network to be run as standalone chain coordinator

### PR DESCRIPTION
**Pull request by MicaiahReid**

### CODE REVIEW

File: components/clarinet-cli/src/chainhooks/types.rs
Comment (MicaiahReid) [inline components/clarinet-cli/src/chainhooks/types.rs:1]: This file wasn't used, so I took the opportunity to delete it

File: components/stacks-network/src/chainhooks.rs
Comment (MicaiahReid) [inline components/stacks-network/src/chainhooks.rs:1]: This `clarinet-cli/src/chainhooks/mod.rs` was only used in `stacks-network`, so I moved it

Comment (MicaiahReid) [reply components/stacks-network/src/chainhooks.rs:1]: When I first made this change, it allowed for some improved code reusability, but I ended up not needing it after all.

So, if anyone doesn't like this being moved I can put it back.

**Review by hugocaillard — DISMISSED** [review]
LGTM
I don't have a lot of experience on stacks-network and stacks-devnet so I think @lgalabru should have a look

File: components/clarinet-cli/src/chainhooks/types.rs
Comment (lgalabru) [reply components/clarinet-cli/src/chainhooks/types.rs:1]: Good call, thanks!

File: dockerfiles/components/stacks-network.dockerfile
Comment (lgalabru) [inline dockerfiles/components/stacks-network.dockerfile:12]: I'm not familiar with distroless, is this a better approach than what we have in the other dockerfiles?

File: components/stacks-network/src/orchestrator.rs
Comment (lgalabru) [inline components/stacks-network/src/orchestrator.rs:177]: Could we please create an issue for tracking these todos? Thanks!

**Review by lgalabru — COMMENTED** [review]
Changes look great, thanks @MicaiahReid! 
Could you also please tell us more about the current status of `clarinet integrate` and `stacks-devnet-js`? Are these 2 features still functional?
Thanks!

File: dockerfiles/components/stacks-network.dockerfile
Comment (MicaiahReid) [reply dockerfiles/components/stacks-network.dockerfile:11]: No idea! I've changed to be the same as what we use for other dockerfiles

File: components/stacks-network/src/orchestrator.rs
Comment (MicaiahReid) [reply components/stacks-network/src/orchestrator.rs:177]: Done: https://github.com/hirosystems/clarinet/issues/1113

**Comment by MicaiahReid** [comment]
@lgalabru, I've confirmed that both `clarinet integrate` and `stacks-devnet-js` still work as expected!

**Comment by CAGS295** [comment]
Hi there! I'm interested in using the event channels from devnetjs and plugging in my own devnet node instead of the orchestrator instantiating it. Basically what I'm trying to figure out is if I can instantiate a node with `clarinet integrate` and then use devnetJS to listen and push events to the node, e.g. mining empty blocks. if you could add a description to this PR, it would help me a lot. Judging from the title and file diffs it looks like this PR may be related to what I would like to implement.

**Comment by MicaiahReid** [comment]
Hey @CAGS295! Usually `clarinet integrate` spins up a few docker containers (bitcoin node, stacks node, stacks api, etc), what we call the network orchestrator (which starts up these processes), and a chain coordinator (which acts as a middleman between the docker containers by forwarding requests between them and sending transactions to fund wallets and run you deployment plan).

This PR adds the ability to _only_ start up the chain coordinator, allowing you to provide a configuration of where the chain coordinator can find the stacks/bitcoin/stacks api nodes. You also provide the chain coordinator with a deployment plan, which it will deploy to those nodes.

Currently, however, this is all very specific to those nodes (that we previously ran as docker containers) being run in kubernetes. We have the host of all of the nodes hardcoded as kubernetes host urls.

This could be made to handle more situations, but it would take some work that isn't currently on the docket for us, but PRs would be welcome once this one is merged!

**Review by lgalabru — APPROVED** [review]
Looking great, thanks @MicaiahReid!

**Pull request merged**
